controllers/complianceeventsapi/server.go

```diff
@@ -471,6 +505,93 @@ func parseQueryArgs(queryArgs url.Values, isCSV bool) (*queryOptions, error) {
 		}
 	}
 
+	parsed, err := setAuthorizedClusters(ctx, db, parsed, userConfig)
+	if err != nil {
+		// ErrNoAccess needs queryOptions
+		return parsed, err
+	}
+
+	return parsed, nil
+}
+
+// getAuthorizedClusterNames verifies that if a cluster filter is provided,
+// the user has access to this filter. If no cluster filter is provided,
+// it sets the cluster filter to all managed clusters the user has access to.
+func setAuthorizedClusters(ctx context.Context, db *sql.DB, parsed *queryOptions,
+	userConfig *rest.Config,
+) (*queryOptions, error) {
+	unAuthorizedClusters := []string{}
+
+	// Get all managedCluster rules
+	allRules, err := getManagedClusterRules(userConfig, nil)
+	if err != nil {
+		return parsed, err
+	}
```

Comment:
Here is an optimization for admin users (common):
```suggestion
	}

	if slices.Contains(allRules["*"], "get") || slices.Contains(allRules["*"], "*") {
		return parsed, nil
	}
```